viadot/sources/s3.py

```diff
@@ -1,266 +1,132 @@
-        ```python
-        from viadot.sources import AzureDataLake
-
-        lake = AzureDataLake(gen=1)
-        lake.exists("tests/test.csv")
-        ```
-
         Returns:
             bool: Whether the paths exists.
         """
-        path = path or self.path
-        return self.fs.exists(path)
-
-    def download(
-        self,
-        to_path: str,
-        from_path: str = None,
-        recursive: bool = False,
-        overwrite: bool = True,
-    ) -> None:
-
-        if overwrite is False:
-            raise NotImplemented(
-                "Currently, only the default behavior (overwrite) is available."
-            )
-
-        from_path = from_path or self.path
-        self.fs.download(rpath=from_path, lpath=to_path, recursive=recursive)
-
-    @add_viadot_metadata_columns
-    def to_df(
-        self,
-        path: str = None,
-        sep: str = "\t",
-        quoting: int = 0,
-        lineterminator: str = None,
-        error_bad_lines: bool = None,
-    ):
-        if quoting is None:
-            quoting = 0
-
-        path = path or self.path
-        url = os.path.join(self.base_url, path)
+        return wr.s3.does_object_exist(boto3_session=self.session, path=path)
 
-        if url.endswith(".csv"):
-            df = pd.read_csv(
-                url,
-                storage_options=self.storage_options,
-                sep=sep,
-                quoting=quoting,
-                lineterminator=lineterminator,
-                error_bad_lines=error_bad_lines,
-            )
-        elif url.endswith(".parquet"):
-            df = pd.read_parquet(url, storage_options=self.storage_options)
-        else:
-            raise ValueError("Only CSV and parquet formats are supported.")
-
-        return df
-
-    def ls(self, path: str = None) -> List[str]:
+    def cp(self, paths: List[str], from_path: str, to_path: str):
         """
-        Returns a list of files in a path.
+        Copies the contents of `from_path` to `to_path`.
 
         Args:
-            path (str, optional): Path to a folder. Defaults to None.
+            paths (List[str]): List of S3 objects paths (e.g. [s3://bucket/dir0/key0, s3://bucket/dir0/key1]).
+            from_path (str, optional): S3 Path for the source directory.
+            to_path (str, optional): S3 Path for the target directory.
         """
-        path = path or self.path
-        return self.fs.ls(path)
 
-    def rm(self, path: str = None, recursive: bool = False):
+        wr.s3.copy_objects(
+            boto3_session=self.session,
+            paths=paths,
+            source_path=from_path,
+            target_path=to_path,
+        )
+
+    def rm(self, path: str):
         """
         Deletes files in a path.
-
         Args:
-            path (str, optional): Path to a folder. Defaults to None.
-            recursive (bool, optional): Whether to delete files recursively.
-                Defaults to False.
+            path (str): Path to a folder.
```

Comment:
Does this only support folders? Also, there's no "recursive" option by any chance?

Reply:
This method supports both files and folders. If the path refers to a folder, it will be removed recursively. Please see the below suggestion:

```suggestion
            path (str): Path to a file or folder to be removed. If the path refers to a folder, it will be removed recursively.
```